salt/modules/disk.py

```diff
@@ -580,3 +586,130 @@ def smart_attributes(dev, attributes=None, values=None):
         smart_attr[attr] = data
 
     return smart_attr
+
+
+@depends(HAS_IOSTAT)
+def iostat(interval=1, count=5, disks=None):
+    '''
+    Gather and return (averaged) IO stats.
+
+    .. versionadded:: Boron
+
+    CLI Example:
+    .. code-block:: bash
+        salt '*' disk.iostat 1 5 disks=sda
+    '''
+    if salt.utils.is_linux():
+        return _iostat_linux(interval, count, disks)
+    elif salt.utils.is_freebsd():
+        return _iostat_fbsd(interval, count, disks)
+
+
+def _iostats_dict(header, stats):
+    '''
+    Transpose collected data, average it, stomp it in dict using header
+
+    Use Decimals so we can properly calc & round, convert to float 'caus' we can't transmit Decimals over 0mq
+    '''
+    stats = [float((sum(stat) / len(stat)).quantize(decimal.Decimal('.01'))) for stat in zip(*stats)]
+    stats = dict(zip(header, stats))
+    return stats
+
+
+def _iostat_fbsd(interval, count, disks):
+    '''
+    Tested on FreeBSD, quite likely other BSD's only need small changes in cmd syntax
+    '''
+    if disks is None:
+        iostat_cmd = 'iostat -xC -w {0} -c {1} '.format(interval, count)
+    elif isinstance(disks, six.string_types):
+        iostat_cmd = 'iostat -x -w {0} -c {1} {2}'.format(interval, count, disks)
+    else:
+        iostat_cmd = 'iostat -x -w {0} -c {1} {2}'.format(interval, count, ' '.join(disks))
+
+    sys_stats = []
+    dev_stats = collections.defaultdict(list)
+    sys_header = []
+    dev_header = []
+    h_len = 1000  # randomly absurdly high
+
+    ret = iter(__salt__['cmd.run_stdout'](iostat_cmd, output_loglevel='quiet').splitlines())
+    for line in ret:
+        if not line.startswith('device'):
+            continue
+        elif not len(dev_header):
+            dev_header = line.split()[1:]
+        while True:
```

Comment:
Could we make this something like `for line in ret`? While I see no obvious case in which this would spin, I tend to have concern about infinite loops in execution modules.